src/consumer/component/consumer-component.js

```diff
@@ -641,14 +634,20 @@ export default class Component {
         specs = path.join(bitDir, bitJson.getSpecBasename());
       }
     }
-    const distPath = path.join(process.cwd(), bitJson.distEntry);
-    const entry = fs.existsSync(distPath) ? distPath : process.cwd();
+    const enrtyPath = path.join(process.cwd(), bitJson.distEntry);
+    const cwd = fs.existsSync(enrtyPath) ? enrtyPath : process.cwd();
+
     const files = componentMap.files;
-    const vinylFiles = Object.keys(files).map(file => new SourceFile(vinylFile.readSync(path.join(consumerPath, files[file]), { base: entry })));
+    const vinylFiles = Object.keys(files).map((file) => {
+      const sourceFile = new SourceFile(vinylFile.readSync(path.join(consumerPath, files[file]), { base: cwd }));
+      sourceFile.distFilePath = path.join(process.cwd(), bitJson.distTarget, sourceFile.relative);
+      return sourceFile;
+    });
     // TODO: Decide about the model represntation
     componentMap.testsFiles.forEach((testFile) => {
-      const file = vinylFile.readSync(path.join(consumerPath, testFile), { base: entry });
+      const file = vinylFile.readSync(path.join(consumerPath, testFile), { base: cwd });
```

Comment:
This should also be inside the load of SourceFile class.
I think the best way is something like this:
Add to load (except from the path and base args) another arg called extendProps which get an array and add those props to the new sourceFile.
Also create loadMany static method on SourceFile class which get an array of paths, a base and the extendProp and return array of sourceFiles